packaging/tests/tas-restapi/src/main/java/org/alfresco/rest/repo/resource/Categories.java

```diff
@@ -0,0 +1,346 @@
+        implements Creator.CategoryCreator, Specifier.CategoriesSpecifier
+    {
+        private final RestWrapper restClient;
+        private RestCategoryModel parent;
+        private final Map<String, RestCategoryModel> categoriesCache;
+
+        protected RepoCategoryCreator(RestWrapper restClient, Map<String, RestCategoryModel> categoriesCache)
+        {
+            super();
+            this.restClient = restClient;
+            this.categoriesCache = categoriesCache;
+        }
+
+        @Override
+        protected CategoryCreator self()
+        {
+            return this;
+        }
+
+        @Override
+        public CategoryCreator category(String name)
+        {
+            return this.withName(name);
+        }
+
+        @Override
+        public CategoryCreator randomCategory()
+        {
+            return this.withRandomName();
+        }
+
+        @Override
+        public CategoryCreator randomCategory(String nameSuffix)
+        {
+            return this.withRandomName(nameSuffix);
+        }
+
+        @Override
+        public MultiCreator.CategoriesCreator categories(String... names)
+        {
+            return new SerialCategoriesCreator(restClient, categoriesCache).withNames(names).underCategory(parent).asUser(user);
+        }
+
+        @Override
+        public MultiCreator.CategoriesCreator randomCategories(String... nameSuffixes)
+        {
+            return new SerialCategoriesCreator(restClient, categoriesCache).withRandomNames(nameSuffixes).underCategory(parent).asUser(user);
+        }
+
+        @Override
+        public MultiCreator.CategoriesCreator randomCategories(int quantity)
+        {
+            return new SerialCategoriesCreator(restClient, categoriesCache).withRandomNames(quantity).underCategory(parent).asUser(user);
+        }
+
+        @Override
+        public MultiCreator.CategoriesCreator nestedCategories(String... names)
+        {
+            return new NestedCategoriesCreator(restClient, categoriesCache).withNames(names).underCategory(parent).asUser(user);
+        }
+
+        @Override
+        public MultiCreator.CategoriesCreator nestedRandomCategories(String... nameSuffixes)
+        {
+            return new NestedCategoriesCreator(restClient, categoriesCache).withRandomNames(nameSuffixes).underCategory(parent).asUser(user);
+        }
+
+        @Override
+        public MultiCreator.CategoriesCreator nestedRandomCategories(int depth)
+        {
+            return new NestedCategoriesCreator(restClient, categoriesCache).withRandomNames(depth).underCategory(parent).asUser(user);
+        }
+
+        @Override
+        public CategoryCreator withRandomName()
+        {
+            return withRandomName(EMPTY);
+        }
+
+        @Override
+        public CategoryCreator withRandomName(String nameSuffix)
+        {
+            withAlias(nameSuffix);
+            return super.withRandomName(CATEGORY_NAME_PREFIX + nameSuffix + "_");
```

Comment:
I think the suffix is used as a prefix here.

Reply:
Yes, this param name is confusing - I changed it